checker/src/visitors.rs

```diff
@@ -1776,11 +1748,13 @@ impl<'analysis, 'compilation, 'tcx, E> MirVisitor<'analysis, 'compilation, 'tcx,
         match &value.expression {
             Expression::CompileTimeConstant(ConstantDomain::Str(s)) => s.clone(),
             _ => {
-                let error = self.session.struct_span_err(
-                    self.current_span,
-                    "this argument should be a string literal, do not call this function directly",
-                );
-                self.emit_diagnostic(error);
+                if self.check_for_errors {
```

Comment:
Not related to the this PR, but I noticed it because of some bewildering panics in the rust compiler.